projects/common/src/navigation/navigation.service.ts

```diff
@@ -89,7 +89,8 @@ export class NavigationService {
           }
         ],
         extras: {
-          skipLocationChange: true // Don't bother showing the updated location, we're going external anyway
+          skipLocationChange: true, // Don't bother showing the updated location, we're going external anyway
+          queryParams: params.queryParams
```

Comment:
Is it possible to encode these into the url we send at this level, so we don't have to worry about stitching together the url downstream?

Reply:
We'll have to stitch it up to url somewhere. Isn't it cleaner to do it here so that the signature for External and Internal navigation params remain clear and uniform? Otherwise, one kind of navigation params will take queryParams as object and another will expect it as part of URL.

Reply:
Yep, sorry I may have been unclear here (which is downstream and which is upstream?!) - I do want it here so callers aren't concerned (which is how it's maintained here), but I was hoping the url we sent to the external nav component would be complete. Since it has already joined all the other pieces like protocol/host/path params, keeping it as a one url rather than treating query params special felt cleaner, letting the nav service deal with the conversion of various inputs to one final url.